main.js

```diff
@@ -153,7 +153,17 @@ async function updateStatus() {
                                 io.emit("update", subreddits[section][subreddit]);
                             else
                                 io.emit("updatenew", subreddits[section][subreddit]);
+                        
+                        // if restricted
+                        }else if(resp.data && resp.data.children[0].data.subreddit_type == "restricted" && subreddits[section][subreddit].status != "restricted"){
 
+                            subreddits[section][subreddit].status = "restricted";
+                            if (firstCheck == false){
+                                io.emit("update", subreddits[section][subreddit]);
+                            }else{
+                                io.emit("updatenew", subreddits[section][subreddit]);
+                            }
+                        
                         } else if (subreddits[section][subreddit].status == "private" && typeof (resp['reason']) == "undefined") {
```

Comment:
i think this needs to be changed to include `"restricted"` as well as `"private"`, otherwise the script will never switch any restricted subreddits back to public

Reply:
this is [what i’ve done](https://github.com/username-is-required/reddark/blob/c3dd2ffe2ed67022ca9d5920a36587ccedd94991/main.js#L216-L218) with this in my fork:

```js
} else if (
    (subreddits[section][subreddit].status == "private" && typeof (data['reason']) == "undefined")
    || (subreddits[section][subreddit].status == "restricted" && data['data'] && data['data']['children'][0]['data']['subreddit_type'] == "public")
) {
```